lib/colorado_lottery.rb

```diff
@@ -0,0 +1,31 @@
+class ColoradoLottery
+  attr_reader :registered_contestants, :winners, :current_contestants
+
+  def initialize
+    @registered_contestants= {}
+    @winners = []
+    @current_contestants = {}
+  end
+
+  def interested_and_18?(contestant, game)
+    if contestant.game_interests.include?(game.name) && contestant.age > 18
+      true
+    else
+      false
+    end
+  end
+
+  def can_register?(contestant, game)
+    if interested_and_18?(contestant, game) == true || contestant.out_of_state? == false
+      true
+    else
+      false
+    end
+  end
+
+  def register_contestant(contestant, game)
+    @registered_contestants[game.name] = contestant
```

Comment:
close, but not quite-- the key is set correctly, but the value needs to be an array of contestants. And setting the value equal doesn't work when you have more than one contestant-- it will always reset to a single contestant. so you'd want to be setting an empty array if the key has never been set before, and then shoveling into the array.

```
  def register_contestant(contestant, game)
    if registered_contestants[game.name] == nil
      registered_contestants[game.name] = []
    end
    registered_contestants[game.name] << contestant
end
```